### PR TITLE
Update/keizai 2.0

### PR DESCRIPTION
**Pull request by nahuelRosas**
## Summary

This PR encompasses a series of refactors and feature enhancements primarily focused on authentication, API integration, and project structure. The improvements are designed to optimize code readability, error handling, and consistency across the codebase.

### Modifications

1. **Refactored Auth and Theming Providers**:
   - Enhanced the structure and management of authentication and theming providers for better modularity.

2. **Introduced Axios Error Handling and API Interfaces**:
   - Implemented comprehensive error handling for Axios requests.
   - Added standardized API interfaces to improve consistency across API calls.

3. **Implemented Auth Context with Key Operations**:
   - Established a context for authentication, handling key operations within the application.

4. **Refactor and Simplification**:
   - Refactored authentication components and simplified validation logic.
   - Updated import paths and corrected typos to ensure consistency.
   - Reorganized CSS classes and updated dependencies for better maintainability.

5. **New Feature: Global Authentication**:
   - Introduced a global authentication mechanism in collaboration with @silviojrcc, enhancing wallet integration and management.
   - Added `idToken` to the sign-in response payload to improve security.

### Changes Overview

- **Files Modified**: Multiple across various aspects of the project
- **Lines Added**: Significant additions focusing on refactoring and feature enhancements
- **Lines Deleted**: Redundant and outdated code removed during refactoring

**Commits Overview:**
- Multiple commits were made on August 1st, 7th, and 8th, 2024, each focusing on specific refactoring or feature enhancements.
- Key contributors: @nahuelRosas and @silviojrcc

### CODE REVIEW

File: src/common/components/auth/ChangePassword.tsx [hunk 1]
```diff
@@ -95,11 +92,11 @@ function ChangePassword() {
 	];
 
 	const onSubmit = async (values: IPasswordReset) => {
-		const { oldPassword, newPassword } = values;
-		await mutate({
-			oldPassword,
-			newPassword,
-		});
+		// const { oldPassword, newPassword } = values;
```
Comment (anarod13) [inline src/common/components/auth/ChangePassword.tsx:95]: Let's remove this comment if code is no longer necessary please

File: src/common/components/auth/ChangePassword.tsx [hunk 2]
```diff
@@ -36,8 +35,8 @@ enum INPUT_PLACEHOLDER {
 }
 
 function ChangePassword() {
-	const { mutation, error } = useChangePasswordMutation();
-	const { mutate, isPending } = mutation;
+	// const { mutation, error } = useChangePasswordMutation();
```
Comment (anarod13) [inline src/common/components/auth/ChangePassword.tsx:38]: Same here!

File: src/pages/auth/ResetPassword.tsx [hunk 3]
```diff
@@ -115,7 +113,7 @@ function ResetPassword() {
 			</div>
 			<div className="flex flex-col mb-4">
 				<Controller
-					control={control}
+					// control={control}
```
Comment (anarod13) [inline src/pages/auth/ResetPassword.tsx:116]: This one to, do we still need it?

**Review by anarod13 — APPROVED** [review]

File: src/common/components/auth/ChangePassword.tsx (hunk 1 above)
Comment (nahuelRosas) [reply src/common/components/auth/ChangePassword.tsx:95]: fixed

File: src/common/components/auth/ChangePassword.tsx (hunk 2 above)
Comment (nahuelRosas) [reply src/common/components/auth/ChangePassword.tsx:38]: fixed

**Review by anarod13 — APPROVED** [review]

**Pull request merged**